### PR TITLE
remove duplicated assignment of case_id

### PR DESCRIPTION
**Pull request by nickpell**
re: https://github.com/dimagi/commcare-hq/pull/13293#discussion_r92798593

case_id is already assigned here: https://github.com/dimagi/commcare-hq/blob/master/corehq/ex-submodules/casexml/apps/case/mock/mock.py#L18

@proteusvacuum @czue

### CODE REVIEW

File: custom/enikshay/nikshay_datamigration/factory.py [hunk 1]
```diff
@@ -33,20 +33,17 @@ def create_cases(self):
 
     def create_person_case(self):
         person_structure = self.person()
-        person_case = self.factory.create_or_update_case(person_structure)[0]
-        person_structure.case_id = person_case.case_id
+        self.factory.create_or_update_case(person_structure)
```
Comment (proteusvacuum) [inline custom/enikshay/nikshay_datamigration/factory.py]: I think you can just remove this call and the one for occurrence too. Both the person and occurrence should get created with this `self.factory.create_or_update_case(episode_structure)`.

Comment (proteusvacuum) [inline custom/enikshay/nikshay_datamigration/factory.py]: see [here](https://github.com/dimagi/commcare-hq/blob/master/custom/enikshay/tests/utils.py#L84-L85) for an example.

Comment (nickpell) [inline custom/enikshay/nikshay_datamigration/factory.py]: updated

**Comment by nickpell** [comment]
✅

**Review by czue — APPROVED** [review]
only reviewed quickly so will let @proteusvacuum merge

File: custom/enikshay/nikshay_datamigration/factory.py [hunk 2]
```diff
@@ -124,14 +107,20 @@ def occurrence(self, outcome):
...
-                self.case_accessor.get_case(self.person().case_id).reverse_indices
-            ])
-            if outcome.pk == occurrence_case.dynamic_case_properties().get('nikshay_id')
-        ), None)
+        if outcome:
+            # TODO - store with correct value
+            kwargs['attrs']['update']['hiv_status'] = outcome.HIVStatus
+
+        try:
+            matching_occurrence_case = next((
```
Comment (proteusvacuum) [inline custom/enikshay/nikshay_datamigration/factory.py]: @nickpell to update this with `case-utils` method

**Review by proteusvacuum — CHANGES_REQUESTED** [review]

File: custom/enikshay/nikshay_datamigration/factory.py (hunk 2 above)
Comment (nickpell) [inline custom/enikshay/nikshay_datamigration/factory.py]: requires https://github.com/dimagi/commcare-hq/pull/14306

File: custom/enikshay/nikshay_datamigration/factory.py [hunk 3]
```diff
@@ -3,7 +3,10 @@
 from casexml.apps.case.const import CASE_INDEX_EXTENSION
 from casexml.apps.case.mock import CaseFactory, CaseStructure, CaseIndex
 from corehq.apps.locations.models import SQLLocation
+from corehq.form_processor.exceptions import CaseNotFound
```
Comment (dimagimon) [inline custom/enikshay/nikshay_datamigration/factory.py]: F401 'CaseNotFound' imported but unused

File: custom/enikshay/nikshay_datamigration/factory.py (hunk 2 above)
Comment (nickpell) [inline custom/enikshay/nikshay_datamigration/factory.py]: updated

File: custom/enikshay/nikshay_datamigration/factory.py [hunk 4]
```diff
@@ -111,21 +113,15 @@ def occurrence(self, outcome):
...
-                ])
-                if self.patient_detail.PregId == occurrence_case.dynamic_case_properties().get('nikshay_id')
-            ), None)
-        except CaseNotFound:
-            matching_occurrence_case = None
-        if matching_occurrence_case:
-            kwargs['case_id'] = matching_occurrence_case.case_id
-            kwargs['attrs']['create'] = False
-        else:
+        if self.person().attrs['create']:
```
Comment (proteusvacuum) [inline custom/enikshay/nikshay_datamigration/factory.py]: nice.

File: custom/enikshay/nikshay_datamigration/factory.py [hunk 5]
```diff
@@ -133,6 +133,7 @@ def episode(self, outcome):
                 'update': {
                     'date_reported': self.patient_detail.pregdate1,  # is this right?
                     'disease_classification': self.patient_detail.disease_classification,
+                    'episode_type': 'confirmed_tb',
```
Comment (proteusvacuum) [inline custom/enikshay/nikshay_datamigration/factory.py]: 👍

**Review by proteusvacuum — APPROVED** [review]
awesome, thanks for updating!

**Pull request merged**